### ISSUE TITLE
New colour scheme for history lines

### ISSUE THREAD

**Issue by rowanj** [rowanj/gitx#245]
I'm considering something a bit less "100% saturation"-ey, with a couple more colours in the mix.

**Comment by rowanj** [comment]
Here's a sample with Ethan Schoonover's "[Solarized](http://ethanschoonover.com/solarized)" accent palette:

![screen shot 2013-08-18 at 7 34 13 pm](https://f.cloud.github.com/assets/47049/982377/9728cb6c-07e9-11e3-9cd9-fe222a629c97.png)

The same, with tighter lane spacing:

![screen shot 2013-08-18 at 7 30 48 pm](https://f.cloud.github.com/assets/47049/982379/9f7d1f34-07e9-11e3-9461-47084eb20985.png)

Original, for comparison:

![screen shot 2013-08-18 at 7 41 55 pm](https://f.cloud.github.com/assets/47049/982385/99275ea0-07ea-11e3-9ae8-be82e82cfd9f.png)

**Comment by pipelineoptika** [comment]
I like the new scheme, but wonder if it will differentiate the lanes as well as the current (horribly gaudy) one.

I am a fan of the compression of the lanes.

I wish there was a way a lane could be highlighted for easier tracing.

**Comment by muhqu** [comment]
:+1: for tighter lane spacing! ...the solarized colors are fine with me, but I don't give them much attention.

**Comment by mengelbrecht** [comment]
:+1:  for Solarized Colors

**Comment by kubark42** [comment]
FWIW, I agree with @pipelineoptika that the horribly gaudy colors are easier to follow. This is because of the increased contrast between neighboring lines. I wonder if that can be improved in the case of the solarized colors by adding a slight shadow or a very fine outline.

For a good example of just how much a difference a black outline makes, look at Google Maps. White text on maps is practically unreadable in certain circumstances, unless you surround it with a bit of black. Then it's good in every circumstance. 

TLDR; A black shadow or outline might provide easier visibility with the solarized color palette.

**Comment by rowanj** [comment]
Shadows are an interesting idea, I'll have a look-see...

**Comment by rowanj** [comment]
Latest revision; added drop shadows as @kubark42 suggested - and I spliced the colour array to reduce the similarity in adjacent lanes.

![screen shot 2013-09-02 at 9 30 34 pm](https://f.cloud.github.com/assets/47049/1067084/f15c1c70-13c3-11e3-9361-d3573c110f5e.png)

The shadows need to be sharp, because the lanes are drawn one cell at a time; and overlapping shadows at every cell boundary do not look good.  The extra space this occupies also makes the tighter lane spacing idea a much tougher call; but I think the overall effect is good.

**Comment by kubark42** [comment]
IMO, this has the best visibility so far. Impressive how quickly you did this!

Totally nitpicking, but at one pixel the black line is thick relative to the colored lines. I wonder if it could be applied with transparency on top of the existing line, so as to reduce the perceived width and still keep the sharpness.

**Comment by muhqu** [comment]
@rowanj I think it would be great if you could make the line-style a user setting and provide some different sets. e.g. bright colors, hard shaddow, soft shadow etc... so that people can try these out, instead of being forced to use some new color scheme.

I personally do not like this hard shadow much, maybe a bit softer does better. But what really help to follow lanes would be some mouse-over-highlighting. e.g. fade all other lanes and commit messages to light gray... I'm sure this is by far not as easy as changing styles, but it would greatly enhance the experience.

**Comment by rowanj** [comment]
The problem with soft shadows is that the lanes are drawn in one-cell-high segments - so they look... bad.

![screen shot 2013-09-02 at 11 03 43 pm](https://f.cloud.github.com/assets/47049/1067453/27691f46-13d0-11e3-97bd-f2f790944d35.png)

There's also practically no connection between segments from one lane to the next, so mouseover highlighting is a Big Problem.  It'd probably be easier from a code standpoint to collapse/hide/fade out the unrelated commits.

I can however lighten up the colour substantially, which has a similar effect.

![screen shot 2013-09-02 at 11 08 47 pm](https://f.cloud.github.com/assets/47049/1067500/45ad7ce4-13d1-11e3-852d-f1d5ca2277b6.png)

Also note:
- thinner border and drop shadow on the commit 'dot'
- no border and shadow on the ref rects
- rounder corners on ref rects
- slight highlight (white shadow) under the ref text to increase contrast
- fixed the missing pixel where a vertical lane hits a corner

Same view in HiDPI:

![screen shot 2013-09-02 at 11 08 59 pm](https://f.cloud.github.com/assets/47049/1067502/49e0d87e-13d1-11e3-900b-48aa6427ceae.png)
